### DISCUSSION TITLE
Refresh all Data Providers

### DISCUSSION THREAD

**Discussion by Kuuchuu** [Budibase/budibase#4554] [Help] [answered]
Is there a way to refresh all data providers on a page? Either automatically every few minutes or with a single button? As far as I can tell you can only refresh a data source with a button if it is nested under the data source.
Thanks in advance!

**Comment by mjashanks** [comment, marked as answer]
Not possible at the moment - I agree that this would be useful.

We do have an issue logged to auto Refresh Data Providers on an interval: https://github.com/Budibase/budibase/issues/3992 - give that an upvote / thumbs-up!

**Reply by Kuuchuu** [reply]
Gotcha, done. For now I had to go ahead and nest all my elements under all of my data providers, which has led to #4557 . The users of the application have to be able to refresh about 4-5 data sources all at once, as multiple refresh buttons will likely lead to confusion amongst the intended users. Thanks for #3992
